### PR TITLE
Add settings to change the map selected by default for a mapset

### LINKED ISSUE #2082: Option to preview the highest difficulty by default when selecting a mapset instead of the easiest one

### PR DESCRIPTION
**Pull request by Illuminati-CRAZ**
Resolves #2082

Instead of selecting the first map in a mapset, the game can now select a map the is closest to/at least/at most some difficulty. Can also prioritize selecting between 4k and 7k in a multimode mapset. The default settings should result in the same map selection behavior as now.

Changed pretty much every seemingly relevant instance of selecting a map, such as `MapManager.Selected.Value = mapset.Maps.First()`, to `MapManager.SelectMapInMapset(mapset)`

Also updated the C# language version of Quaver.Shared to the latest (C#10) and the launch.json file for vscode debugging.

### CODE REVIEW

**Comment by Swan** [comment]
How about instead of doing this, we use the player's local/online rating to get the approximate difficulty they should play, and set the map based on that? 

Just seems a lot simpler than having the user set two more options (adding complexity) for arguably the same result.

**Comment by Illuminati-CRAZ** [comment]
I guess you're right, but imo I would leave the target difficulty in case they either don't like the auto determined difficulty (or just wanna auto-select the lowest or highest diff), and I think the gamemode prioritization would be nice for most people who mainly play one keymode.

**Comment by Swan** [comment]
Both are fine with me. Target difficulty should just be an override of the default behavior

**Comment by Illuminati-CRAZ** [comment]
DIfficulty suggester isn't tested, can't test with online profile, but the code was modelled off of the difficulty suggester on download screen. Ideally it would use the local profile when offline, but idk how to access the local stats.

**Comment by IceDynamix** [comment]
You can check how the F4 profile in the song selection initializes the different profiles for the table here https://github.com/Quaver/Quaver/blob/ui-redesign/Quaver.Shared/Screens/Selection/UI/Profile/ProfileSelectionDropdown.cs

File: Quaver.Shared/Screens/Options/Items/Custom/OptionsItemSuggestDifficulty.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+                if (!OnlineManager.Connected || OnlineManager.Self == null)
+					return;
+
+                var rating = OnlineManager.Self.Stats[GameMode.Keys4].OverallPerformanceRating;
+                var diff = rating / 20f;
+				ConfigManager.PrioritizedMapDifficulty4K.Value = (int)(diff * 10);
+
+                rating = OnlineManager.Self.Stats[GameMode.Keys7].OverallPerformanceRating;
+                diff = rating / 20f;
+                ConfigManager.PrioritizedMapDifficulty7K.Value = (int)(diff * 10);
```
Comment (Swan) [inline Quaver.Shared/Screens/Options/Items/Custom/OptionsItemSuggestDifficulty.cs:46]: Try using [UserProfileDatabaseCache.Selected](https://github.com/Quaver/Quaver/blob/7ac14c2c6526a4bd89be01e28412a2e591aa7e11/Quaver.Shared/Database/Profiles/UserProfileDatabaseCache.cs) instead?

**Review by Swan — CHANGES_REQUESTED** [review]

File: Quaver.Shared/Quaver.Shared.csproj [hunk 2]
```diff
@@ -1,4 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <LangVersion>latest</LangVersion>
+  </PropertyGroup>
```
[the remark this answers is not in the document]
Comment (IceDynamix) [reply Quaver.Shared/Quaver.Shared.csproj:4]: Agree. If anything, this should be discussed and implemented in a separate issue/PR.

[the remark this answers is not in the document]
Comment (Illuminati-CRAZ) [reply Quaver.Shared/Quaver.Shared.csproj:4]: Will revert, it looks like there could also potentially be incompatibilities with the target framework that I didn't consider

**Review by Swan — APPROVED** [review]

**Pull request merged**